diff -up swig-4.1.0/Lib/perl5/perlhead.swg.omv~ swig-4.1.0/Lib/perl5/perlhead.swg
--- swig-4.1.0/Lib/perl5/perlhead.swg.omv~	2022-11-29 00:15:27.490200845 +0100
+++ swig-4.1.0/Lib/perl5/perlhead.swg	2022-11-29 00:15:42.141302547 +0100
@@ -16,8 +16,10 @@ extern "C" {
 #include "XSUB.h"
 
 #if __GNUC__ >= 10
+#if defined(__cplusplus)
 #pragma GCC diagnostic pop
 #endif
+#endif
 
 /* PERL_REVISION was added in Perl 5.6. */
 #if !defined PERL_REVISION || (PERL_REVISION-0 == 5 && PERL_VERSION-0 < 8)
