diff -up swig-4.1.1/Lib/ocaml/ocamlrundec.swg.1~ swig-4.1.1/Lib/ocaml/ocamlrundec.swg
--- swig-4.1.1/Lib/ocaml/ocamlrundec.swg.1~	2022-11-30 08:35:05.000000000 +0100
+++ swig-4.1.1/Lib/ocaml/ocamlrundec.swg	2023-04-12 03:36:27.025288957 +0200
@@ -9,6 +9,8 @@
 #include <stdlib.h>
 
 #ifdef __cplusplus
+#include <memory>
+
 #define SWIGEXT extern "C"
 SWIGEXT {
 #else
@@ -84,7 +86,7 @@ SWIGEXT {
 #undef CAMLreturn0
 #endif
 #define CAMLreturn0 do{ \
-  caml_local_roots = caml__frame; \
+  *caml_local_roots_ptr = caml__frame; \
   return; \
 }while (0)
 
@@ -93,12 +95,12 @@ SWIGEXT {
 #endif
 #define CAMLreturn(result) do{ \
   caml_value_t caml__temp_result = (result); \
-  caml_local_roots = caml__frame; \
+  *caml_local_roots_ptr = caml__frame; \
   return (caml__temp_result); \
 }while(0)
 
 #define CAMLreturn_type(result) do{ \
-  caml_local_roots = caml__frame; \
+  *caml_local_roots_ptr = caml__frame; \
   return result; \
 }while(0)
 
