diff -up ./configure.ac.pylib ./configure.ac
--- ./configure.ac.pylib	2014-03-16 23:49:54.000000000 +0200
+++ ./configure.ac	2014-03-22 19:20:04.519686519 +0200
@@ -642,11 +642,13 @@ else
     fi
     AC_MSG_RESULT($PYLIBDIR)
 
+    PYLIB=`$PYTHON -c "import sys; print sys.path[[1]]"`
+
     # Set the include directory
 
     AC_MSG_CHECKING(for Python header files)
     if test -r $PYPREFIX/include/$PYVERSION/Python.h; then
-      PYINCLUDE="-I$PYPREFIX/include/$PYVERSION -I$PYEPREFIX/$PYLIBDIR/$PYVERSION/config"
+      PYINCLUDE="-I$PYPREFIX/include/$PYVERSION -I$PYLIB/config"
     fi
     if test -z "$PYINCLUDE"; then
       if test -r $PYPREFIX/include/Py/Python.h; then
@@ -657,13 +659,6 @@ else
 
     # Set the library directory blindly.   This probably won't work with older versions
     AC_MSG_CHECKING(for Python library)
-    dirs="$PYVERSION/config $PYVERSION/$PYLIBDIR python/$PYLIBDIR"
-    for i in $dirs; do
-      if test -d $PYEPREFIX/$PYLIBDIR/$i; then
-        PYLIB="$PYEPREFIX/$PYLIBDIR/$i"
-        break
-      fi
-    done
     if test -z "$PYLIB"; then
       AC_MSG_RESULT(Not found)
     else
