--- swig-3.0.6/configure.ac.pylib~	2015-07-28 13:50:47.779084837 +0200
+++ swig-3.0.6/configure.ac	2015-07-28 13:58:32.808195207 +0200
@@ -613,7 +613,7 @@ if test x"${PYBIN}" = xno -o x"${with_al
 else
   # First figure out the name of the Python executable
   if test "x$PYBIN" = xyes; then
-    AC_CHECK_PROGS(PYTHON, [python python2.8 python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python2.0 python1.6 python1.5 python1.4])
+    AC_CHECK_PROGS(PYTHON, [python python2.8 python2.7 python2 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python2.0 python1.6 python1.5 python1.4])
   else
     PYTHON="$PYBIN"
   fi
@@ -682,10 +682,11 @@ else
       AC_MSG_RESULT($PYLIBDIR)
 
       # Set the include directory
+      PYLIB=`$PYTHON -c "import sys; print(sys.path[[1]])"`
 
       AC_MSG_CHECKING(for Python header files)
       if test -r $PYPREFIX/include/$PYVERSION/Python.h; then
-        PYINCLUDE="-I$PYPREFIX/include/$PYVERSION -I$PYEPREFIX/$PYLIBDIR/$PYVERSION/config"
+        PYINCLUDE="-I$PYPREFIX/include/$PYVERSION -I$PYLIB/config"
       fi
       if test -z "$PYINCLUDE"; then
         if test -r $PYPREFIX/include/Py/Python.h; then
